packages/echo-core/src/types/panel.ts

```diff
@@ -18,6 +18,7 @@ export interface Panel {
     key: string;
     label: string;
     icon: Icon;
+    isToggleButtonVisible?: (() => boolean) | boolean;
```

Comment:
I think this should only be a function. Both a function and a boolean makes it hard to use and understand for other modules.

Reply:
When I used this new property in echopedia with type Panel, VSCode provided enough hint to see that this is either a boolean or a function returning a boolean value:
![image](https://user-images.githubusercontent.com/13872908/150086326-3dca2f4a-2960-4553-b8fc-44e29d04890d.png)

Based on this I think it's easy to use and understand what is going on, when registering panels.

My reason to make it possible to pass just a primitive value is performance based: if one passes a function, that function will be called many times during UI changes & re-rendering. Calling a function is more costly then just checking a primitive value.
So I wanted to leave a possibility to pass just a primitive value - if that is enough in a given scenario - to improve performance of the app.

Reply:
This is [premature optimization code smell](https://www.google.com/search?q=code+smell+premature+optimization&rlz=1C1GCEA_enNO867NO867&oq=code+smell+premature+optimization&aqs=chrome..69i57j69i64.4286j0j7&sourceid=chrome&ie=UTF-8)

Having to check if it's a function or boolean complicates the code and is not worth it.